DOCS-4375 USM updates for release (#15971)
DataDog/documentation: content/en/universal_service_monitoring/_index.md

  text: "Read about the Service Map"
---

## Overview

> Universal Service Monitoring (USM) provides visibility into your service health metrics universally across your entire stack _without having to instrument your code_. It relies solely on the presence of a configured Datadog Agent and [Unified Service Tagging][1], and brings performance data about your uninstrumented services into views such as the Service Catalog and Service Map. USM also works with [Deployment Tracking][2], Monitors, Dashboards, and SLOs.

Reviewer: I will add links to these additional doc sections in a follow up PR.

> Universal Service Monitoring (USM) provides visibility into your service health metrics universally across your entire stack _without having to instrument your code_. It relies solely on the presence of a configured Datadog Agent and [Unified Service Tagging][1], and brings performance data about your uninstrumented services into APM views such as the Service Catlog, Services List, and Service Map. USM also works with [Deployment Tracking][2], Monitors, Dashboards, and SLOs.

Author: ```suggestion
Universal Service Monitoring (USM) provides visibility into your service health metrics universally across your entire stack _without having to instrument your code_. It relies solely on the presence of a configured Datadog Agent and [Unified Service Tagging][1], and brings performance data about your uninstrumented services into APM views such as the Service Catalog, Services List, and Service Map. USM also works with [Deployment Tracking][2], Monitors, Dashboards, and SLOs.
```

> Universal Service Monitoring (USM) provides visibility into your service health metrics universally across your entire stack _without having to instrument your code_. It relies solely on the presence of a configured Datadog Agent and [Unified Service Tagging][1], and brings performance data about your uninstrumented services into APM views such as the Service Catalog, Services List, and Service Map. USM also works with [Deployment Tracking][2], Monitors, Dashboards, and SLOs.

Reviewer 2: ```suggestion
Universal Service Monitoring (USM) provides visibility into your service health metrics universally across your entire stack _without having to instrument your code_. It relies solely on the presence of a configured Datadog Agent and [Unified Service Tagging][1], and brings performance data about your uninstrumented services into views such as the Service Catalog and Service Map. USM also works with [Deployment Tracking][2], Monitors, Dashboards, and SLOs.
```

Reviewer 2: Service Catalog is eventually replacing Service List

{{< img src="universal_service_monitoring/usm-demo.mp4" alt="Video demonstrating Universal Service Monitoring. An overview of a service is accessed by clicking on a service on the Service Map and selecting View service overview." video="true" >}}

> ### Supported versions and compatibility

Reviewer: ```suggestion

## Setup

### Supported versions and compatibility
```

Required Agent version

> : Universal Service Monitoring requires that the Datadog Agent installed alongside your containerized service be at least version 6.40/7.40.

Reviewer: ```suggestion
: Universal Service Monitoring requires that the Datadog Agent installed alongside your containerized service be at least version 6.40 or 7.40.
```

Supported platforms
: Linux kernel 4.14 and greater<br/>
CentOS or RHEL 8.0 and greater<br/>

> Windows IIS

Reviewer 2: Windows 2012 R2 and above

> Windows IIS

Author: ```suggestion
IIS on Windows 2012 R2 and greater
```

> : Linux kernel 4.14 and greater<br/>

Reviewer: ```suggestion
: Linux Kernel 4.14 and greater<br/>
```

Supported application-layer protocols

> : HTTP<br/>

Reviewer 3: For HTTPS (below this line), let's add GnuTLS alongside OpenSSL?

Author: ```suggestion
: HTTP<br/>
HTTPS with GnuTLS or Open SSL
```
Is this what you mean, @sunhay ?

Reviewer 2: Looks good!

> ### Prerequisites

Reviewer 2: Could we add "containers" (Linux) as a prereq in this section? We've seen some customers try setting this up on non-container Linux setups

Author: In line 53 we talk about non-container setups. Are they supported (i.e., do you just mean to indicate we don't support serverless) or should we remove line 53 and claim support only for containerized linux services?

Reviewer 2: I think we should remove line 53 to avoid this confusion altogether. We don't officially support non-container setups yet (e.g. baremetal on a VM or serverless)

Reviewer 2: So we should make it clear, potentially under `Supported Platforms` that containers are required

> - Datadog Agent 6.40/7.40 or higher is installed alongside your service. Installing a tracing library is _not_ required.

Reviewer 3: Do we need this? It's duplicated with whats above and that section seems to be clearer and easier to read in **Supported versions and compatibility**

> - Datadog Agent 6.40/7.40 or higher is installed alongside your service. Installing a tracing library is _not_ required.

Author: ```suggestion
- Datadog Agent is installed alongside your service. Installing a tracing library is _not_ required.
```

### Prerequisites

> - Your service is running in a container

Reviewer: ```suggestion
- Your service is running in a container.
```

- Datadog Agent is installed alongside your service. Installing a tracing library is _not_ required.

> - [Unified Service Tagging][1] tag for `env` has been applied to your deployment. The `service` and `version` tags are optional.

Reviewer: ```suggestion
- The `env` tag for [Unified Service Tagging][1] has been applied to your deployment. The `service` and `version` tags are optional.
```

```

{{% /tab %}}

> {{% tab "Windows" %}}

Reviewer 3: We should get the instructions for ECS and get it added as well

> **For services running on IIS:**

Reviewer 2: Should we call out what a customer should do if they're not using IIS? Opening a support ticket would be a good call-to-action

Author: If we do, we should instead put that call to action in the "Supported versions" section -- if you don't see your platform listed, contact support. And you'd need to update the internal confluence page on what to do when the customer calls support from this CTA. What do you think?

Reviewer 2: Sounds good to me!

```

{{% /tab %}}
{{% tab "Windows" %}}

**For services running on IIS:**

> 1. Install the [Datadog Agent][1] (version 6.40/7.40 or above) with the network driver component enabled. During installation pass `ADDLOCAL="MainApplication,NPM"` to the `msiexec` command, or select "Network Performance Monitoring" when running the Agent installation through the GUI.

Reviewer: ```suggestion
1. Install the [Datadog Agent][1] (version 6.40 or 7.40 and later) with the network driver component enabled. During installation, pass `ADDLOCAL="MainApplication,NPM"` to the `msiexec` command, or select **Network Performance Monitoring** when running the Agent installation through the UI.
```

To update the service's name, use [unified service tagging][1].

{{< img src="universal_service_monitoring/automatic-service-tagging.png" alt="When Datadog automatically detects your services, the tag used for this is shown on the top of the service page" style="width:80%;" >}}

## Exploring your services

> After you configure the Agent, wait about five minutes for your service to appear in the APM Service Catalog. Click the service to see the APM service details page. An operation name of `universal.http.server` or `universal.http.client` in the upper left indicates that the service telemetry comes from Universal Service Monitoring.

Reviewer 2: ```suggestion
After you configure the Agent, wait about five minutes for your service to appear in the Service Catalog. Click the service to see the service details page. An operation name of `universal.http.server` or `universal.http.client` in the upper left indicates that the service telemetry comes from Universal Service Monitoring.
```

Reviewer 2: Want to avoid confusion that there's a dependency on APM here

> To update the service's name, use [unified service tagging][1].

Reviewer: ```suggestion
To update the service's name, set up [Unified Service Tagging][1].
```
